packages/preferences-controller/src/PreferencesController.ts

```diff
@@ -319,6 +351,26 @@ export class PreferencesController extends BaseController<
   setIsIpfsGatewayEnabled(isIpfsGatewayEnabled: boolean) {
     this.update({ isIpfsGatewayEnabled });
   }
+
+  /**
+   * A setter for the user allow to be fetched IPFS content
+   *
+   * @param chainId - On hexadecimal format to enable the incoming transaction network
+   * @param isIncomingTransactionNetworkEnable - true to enable incoming transactions
+   */
+  setEnableNetworkIncomingTransactions(
+    chainId: EtherscanSupportedHexChainId,
+    isIncomingTransactionNetworkEnable: boolean,
+  ) {
+    if (Object.values(ETHERSCAN_SUPPORTED_CHAIN_IDS).includes(chainId)) {
+      this.update({
+        showIncomingTransactions: {
+          ...this.state.showIncomingTransactions,
+          [chainId]: isIncomingTransactionNetworkEnable,
+        },
+      });
+    }
```

Comment:
The check on line 365 is probably redundant because of the compile-time check on line 362. But having a run-time check here might not be a bad idea as long as it doesn't get too expensive. 

```suggestion
    this.update({
      showIncomingTransactions: {
        ...this.state.showIncomingTransactions,
        [chainId]: isIncomingTransactionNetworkEnable,
      },
    });
```

Reply:
Yeah, I agree, since the list is not that big for now I will let this run-time check, since it doesn't seem to get that expensive! I will go ahead and merge it, let me know if you think otherwise and I will open a PR to address the change